### PR TITLE
Add HTTP request and duration metrics

### PR DESCRIPTION
**Pull request by povilasv**
Adding more metrics :)

I have pushed my own docker.io/povilasv/sealed-secrets-controller:latest image and will test it more :)

Tested it more. Looks fine.

Example of new metrics:

```

sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="0.01",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
-- | --
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="0.025",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="0.05",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="0.1",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="0.25",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="0.5",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="1",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="10",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="2.5",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_bucket{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",le="5",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_count{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"} | 4
sealed_secrets_controller_request_duration_seconds_sum{hostname="rock64-0",instance="10.200.0.51:8080",job="kubernetes-service-endpoints",method="get",name="sealed-secrets-controller",namespace="kube-system",path="/v1/cert.pem",pod="sealed-secrets-controller-779566744-jp6rk"}

```

### CODE REVIEW

**Comment by povilasv** [comment]
@mkmik hope you are doing well!

Any chance you have time to take a look? Would love to get some feedback from you!

**Comment by kskewes** [comment]
Cool. Looking on phone and haven't looked at the Prometheus docs for the handler wrapper function.

Do you use Grafana? 
If so would you like to add a couple of panels to the dashboard? 
Perhaps using rate of requests for counter and then p50, p90, p99 quantiles for duration?

**Comment by mkmik** [comment]
@povilasv  could you please rebase? I switched to GH actions so that also PRs can run integration tests

File: cmd/controller/metrics.go [hunk 1]
```diff
@@ -55,6 +58,25 @@ var (
...
+			Name:      "api_requests_total",
+			Help:      "A counter for requests to the wrapped handler.",
+		},
+		[]string{"path", "code", "method"},
+	)
+
+	duration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricNamespace,
+			Name:      "request_duration_seconds",
```
Comment (mkmik) [inline cmd/controller/metrics.go:74]:
```suggestion
			Name:      "http_request_duration_seconds",
```

File: cmd/controller/metrics.go [hunk 2]
```diff
@@ -55,6 +58,25 @@ var (
 		Name:      "condition_info",
 		Help:      "Current SealedSecret condition status. Values are -1 (false), 0 (unknown or absent), 1 (true)",
 	}, []string{labelNamespace, labelName, labelCondition})
+
+	counter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "api_requests_total",
```
Comment (mkmik) [inline cmd/controller/metrics.go:65]: https://prometheus.io/docs/practices/naming/

```suggestion
			Name:      "http_requests_total",
```

**Comment by povilasv** [comment]
Rebased and fixed the comments :) 

@kskewes I do, but I haven't worked on it yet

File: cmd/controller/metrics.go [hunk 3]
```diff
@@ -55,6 +58,25 @@ var (
 		Name:      "condition_info",
 		Help:      "Current SealedSecret condition status. Values are -1 (false), 0 (unknown or absent), 1 (true)",
 	}, []string{labelNamespace, labelName, labelCondition})
+
+	counter = prometheus.NewCounterVec(
```
Comment (kskewes) [inline cmd/controller/metrics.go:62]:
```suggestion
	httpRequestsTotal = prometheus.NewCounterVec(
```

File: cmd/controller/metrics.go [hunk 4]
```diff
@@ -55,6 +58,25 @@ var (
...
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "http_requests_total",
+			Help:      "A counter for requests to the wrapped handler.",
+		},
+		[]string{"path", "code", "method"},
+	)
+
+	duration = prometheus.NewHistogramVec(
```
Comment (kskewes) [inline cmd/controller/metrics.go:71]:
```suggestion
	httpRequestDurationSeconds = prometheus.NewHistogramVec(
```

File: cmd/controller/metrics.go [hunk 5]
```diff
@@ -64,6 +86,8 @@ func init() {
 	prometheus.MustRegister(unsealRequestsTotal)
 	prometheus.MustRegister(unsealErrorsTotal)
 	prometheus.MustRegister(conditionInfo)
+	prometheus.MustRegister(counter)
```
Comment (kskewes) [inline cmd/controller/metrics.go:89]:
```suggestion
	prometheus.MustRegister(httpRequestsTotal)
```

File: cmd/controller/metrics.go [hunk 6]
```diff
@@ -64,6 +86,8 @@ func init() {
...
+	prometheus.MustRegister(duration)
```
Comment (kskewes) [inline cmd/controller/metrics.go:90]:
```suggestion
	prometheus.MustRegister(httpRequestDurationSeconds)
```

File: cmd/controller/metrics.go (hunk 3 above)
Comment (kskewes) [reply cmd/controller/metrics.go:62]: We're likely to have more counters and histograms later.

**Review by kskewes — CHANGES_REQUESTED** [review]

**Comment by povilasv** [comment]
@kskewes thanks for taking a look :1st_place_medal: . Applied you suggestions ;)

**Review by mkmik — APPROVED** [review]

**Pull request merged**